### ISSUE TITLE
feature "delete channel/account allowed = YES/NO" as addon or included as default?

### ISSUE THREAD

**Issue by lazyadmin111** [redmatrix/hubzilla#830]
Hey there,

Some of my users did delete their account and/or channel on my server because they wanted to annoy me. And unfortunately it is not that easy to restore it again. Why they do that? -- well, e.g. one "friend" posted just porn links to my posts (his gf left him and he tried to vent his anger on me...), and so I forbid him to comment or publish anything on my blog. The result was, that he then deleted his account and 2 months later now wishes it back.

This was not the first time, people see that privacy things matter to me, and how I like hubzilla, and thus if they have some reason, real or not, to be angry at me they just delete the account. They often think that it's as easy to restore the "friendship" like it is on facebook. There ppl delete their "friends" to express their angryness and then, after they calm down, one of both sends again a "friendship request". 

In short, deleting an account or channel is used unfortunately by some users on my server to express sth. Unfortunately this causes me additional work, I have to fiddle around in the database to restore their account and thus I would like to prevent this.

I already modified the code of my hubzilla installation such that removing the channel doesn't work any more. Now I would like to integrate that behavior better in the existing project, and I wonder if I should just write an addon for hubzilla or you would also want to see this as an configurable option in your project.

Please let me now,
thanks for your great work!

**Comment by git-marijus** [comment]
I personally consider prohibiting deletion of accounts an antifeature and would not want this to be in hubzilla core or addons.

**Comment by lazyadmin111** [comment]
sure, I will test that in the eve.

**Comment by zotlabs** [comment]
This has come up before. 

>  **WARNING**: This account and all its channels will be completely removed from the network. **This action is permanent and can not be undone!** 

We then ask for your password (which can't have been changed in the past 24 hours) to verify your intent. So I'm sympathetic, but we aren't in the habit of lying to folks. We tell them clearly what to expect from this action. 

I would be OK with a config option which hid the buttons to do these things, but one can still go directly to the appropriate URL and initiate the action. There have been huge shit-fights in the federated web over services that **don't** let you remove your account or don't remove it satisfactorily; so it's important that we provide this ability.

**Comment by albigro** [comment]
I'm just an user more or less, but ...
No one can be saved from doing stupid things to himself, just by not allowing it. Anyone has the right to do so and to live with the consequences, to learn.
If i make an update and a warning tells me to be careful, make backups, safe the database, ... and i think i'm lazy, come on, i just do it without a backup,...  Next time i think twice about being lazy, if messed up something ones.

**Comment by lazyadmin111** [comment]
Well, I am less concerned about the loss of the user, but more about the additional time to handle ppl like these, which don't care about their posts but just want to annoy me.

I think the additional option to hide the buttons would be great. Then it wouldn't make it so easy to delete the channel/account. I have no problem with the appropriate URL, to let serious users delete anything they want to.

**Comment by lazyadmin111** [comment]
Maybe one idee to easy that point would be to differentiate between "delete content" and delete the account and channel itself.

**Comment by ilu33** [comment]
Actually this proposal makes sense for closed usergroups. If hubzilla is used as a communication tool in a club, school or other group where organized communication is mandatory I can imagine scenarios where you couldn't allow any member to opt out of communication. People not willing to communicate would have to leave the club instead. Luckily account deletion does not work on our hub and I hope it stays that way.
